src/swap.ts

```diff
@@ -13,31 +19,85 @@ export class Swap {
     readonly self: string
   ) {}
 
-  public async accept(params: TryParams) {
-    return await this.tryExecuteAction("accept", params);
+  /**
+   * Looks for and executes the accept action of this {@link Swap}.
+   * If the {@link Swap} is not in the right state this call will throw a timeout exception.
+   *
+   * @param tryParams Controls at which stage the exception is thrown.
+   */
+  public async accept(tryParams: TryParams): Promise<void> {
+    await this.tryExecuteSirenAction<void>("accept", tryParams);
   }
 
-  public async decline(params: TryParams) {
-    return await this.tryExecuteAction("decline", params);
+  /**
+   * Looks for and executes the decline action of this {@link Swap}.
+   * If the {@link Swap} is not in the right state this call will throw a timeout exception.
+   *
+   * @param tryParams Controls at which stage the exception is thrown.
+   */
+  public async decline(tryParams: TryParams): Promise<void> {
+    await this.tryExecuteSirenAction<void>("decline", tryParams);
   }
 
-  public async deploy(params: TryParams) {
-    const response = await this.tryExecuteAction("deploy", params);
+  /**
+   * Looks for and executes the deploy action of this {@link Swap}.
+   * If the {@link Swap} is not in the right state this call will throw a timeout exception.
```

Comment:
Does `@throw` exist in typedoc? If so, it should be preferred.

Reply:
We don't have a dedicated exception though, maybe we should add one!

Reply:
added to list of improvements here: https://github.com/comit-network/comit-js-sdk/issues/96